trace/noop.go

```diff
@@ -43,6 +43,12 @@ func (t noopTracer) Start(ctx context.Context, name string, _ ...SpanStartOption
 		// span is likely already a noopSpan, but let's be sure
 		span = noopSpanInstance
 	}
+
+	if ctx == nil {
```

Comment:
Shouldn't we put the condition at the beginning of the method (before the `SpanFromContext(ctx)` call?)